storage/store.go

```diff
@@ -1264,8 +1264,10 @@ func (s *Store) ExecuteCmd(ctx context.Context, args proto.Request) (proto.Respo
 	trace := tracer.FromCtx(ctx)
 	// If the request has a zero timestamp, initialize to this node's clock.
 	header := args.Header()
-	if err := verifyKeys(header.Key, header.EndKey, proto.IsRange(args)); err != nil {
-		return nil, err
+	if args.Method() != proto.Batch { // TODO(tschottdorf)
```

Comment:
I moved the verification to each individual request in `checkBatchRequest` which is called from `Replica`. So the TODO is gone (it was that `verifyKeys` on a `BatchRequest` doesn't make sense).